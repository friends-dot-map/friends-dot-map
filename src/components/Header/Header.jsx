import NavButton from '../NavButton/NavButton';

export default function Header() {
  return (
    <header className="flex items-center gap-7 justify-between tracking-wider text-left bg-white ring-tint bg-opacity-40 w-full p-3 rounded-md">
      <h1 className="text-dark font-semibold font-cursive text-3xl  ">
        <span className="text-teal">friends</span>.
        <span className="text-orange">map</span>(🗺️)
      </h1>
      <NavButton />
    </header>
  );
}
