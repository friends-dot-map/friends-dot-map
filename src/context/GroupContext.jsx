import { createContext, useContext, useState, useEffect } from 'react';
import { getAllProfiles } from '../services/profiles';

export const GroupContext = createContext();

const GroupProvider = ({ children }) => {
  const [group, setGroup] = useState([]);

  useEffect(() => {
    const fetchGroup = async () => {
      try {
        const data = await getAllProfiles();
        setGroup(data);
      } catch (error) {
        throw error('something went horribly wrong');
      }
    };
    fetchGroup();
  }, []);
  const groupValue = { group, setGroup };
  return (
    <GroupContext.Provider value={groupValue}>{children}</GroupContext.Provider>
  );
};

const useGroup = () => {
  const context = useContext(GroupContext);

  if (context === undefined) {
    throw new Error('useGroup must be used within a GroupProvider');
  }

  return context;
};

export { GroupProvider, useGroup };
