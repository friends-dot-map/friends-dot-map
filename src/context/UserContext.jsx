import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
