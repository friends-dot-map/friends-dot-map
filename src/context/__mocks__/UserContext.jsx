import { createContext, useContext, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
