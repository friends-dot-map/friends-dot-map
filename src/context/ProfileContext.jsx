import { createContext, useContext, useState, useEffect } from 'react';
import { getProfile } from '../services/profiles';
import { useUser } from './UserContext';
import { formatDate } from '../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [loading, setLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = await getProfile(user.email);
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    loading,
    setLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
