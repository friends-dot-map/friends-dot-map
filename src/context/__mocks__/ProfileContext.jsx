import { createContext, useContext, useState, useEffect } from 'react';
import { useUser } from '../UserContext';
import { formatDate } from '../../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ mockProfile, children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [profileLoading, setProfileLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = mockProfile || {};
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setProfileLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    profileLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
