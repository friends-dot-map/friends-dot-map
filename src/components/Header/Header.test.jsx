import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ProfileProvider } from '../../context/ProfileContext';
import { UserProvider } from '../../context/UserContext';

import Header from './Header';

test('should render the header', () => {
  const {container} = render(
    <UserProvider>
      <ProfileProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </ProfileProvider>
    </UserProvider>
  );
  

  expect(container).toMatchSnapshot();
});
