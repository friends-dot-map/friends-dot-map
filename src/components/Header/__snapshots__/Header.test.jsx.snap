// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header 1`] = `
<div>
  <header
    class="flex items-center gap-7 justify-between tracking-wider text-left bg-white ring-tint bg-opacity-40 w-full p-3 rounded-md"
  >
    <h1
      class="text-dark font-semibold font-cursive text-3xl  "
    >
      <span
        class="text-teal"
      >
        friends
      </span>
      .
      <span
        class="text-orange"
      >
        map
      </span>
      (üó∫Ô∏è)
    </h1>
    <div
      class="relative inline-block text-left"
    >
      <div>
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="bg-white/0 inline-flex justify-center w-full rounded-md shadow-sm px-4 py-2 text-base font-medium text-dark hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500"
          id="headlessui-menu-button-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="h-10 w-10"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</div>
`;
